### PR TITLE
Services NodePort E2Es: Reset shared variables before each test

### PR DESCRIPTION
**Pull request by kyrtapz**
Reset shared variables before each test to avoid cleanup issues.
Fixes: https://github.com/ovn-org/ovn-kubernetes/issues/4733.

### CODE REVIEW

File: test/e2e/service.go [hunk 1]
```diff
@@ -740,6 +740,7 @@ var _ = ginkgo.Describe("Services", func() {
 					}
 				}
 			}
+			nodeIPs = make(map[string]map[int]string)
```
Comment (dceara) [inline test/e2e/service.go:743]: Shouldn't we do this also just before the `framework.Failf("failed to remove ip address %s from node %s, err: %q", ip, nodeName, err)` call above?

**Comment by dceara** [comment]
The commit log probably needs a
```
Fixes: https://github.com/ovn-org/ovn-kubernetes/issues/4733
```

**Comment by dceara** [comment]
Also, @tssurya can we please also cherry pick this to the `release-1.0` branch after it gets merged to master?  We hit this in ovn-orv/ovn CI too.

File: test/e2e/service.go (hunk 1 above)
Comment (dceara) [reply test/e2e/service.go:743]: I meant when we fail.  `framework.Failf(..)` will break out of the loop, right?  But if we don't reset the map there we might cause other tests in the suite (that are executed after the currently failing one) to fail too.

Comment (npinaeva) [reply test/e2e/service.go:743]: came here to post the same fix :)
I think the most robust way would be to move 
```
nodeIPs = make(map[string]map[int]string)
// and maybe this one for company
targetSecondaryNode = node{
			name: "egressSecondaryTargetNode-allowed",
		}
```
to BeforeEach?

Comment (kyrtapz) [reply test/e2e/service.go:743]: Good point, I will wait for the control-plane tests to pass and will change this to:
```
ginkgo.BeforeEach(func() {
	nodeIPs = make(map[string]map[int]string)
})
```
I think it makes more sense logically anyway.

Comment (kyrtapz) [reply test/e2e/service.go:743]: Actually, I will reset all shared vars to avoid any future issues.

Comment (kyrtapz) [reply test/e2e/service.go:743]: All control-plane lanes are green, re-pushing.

**Review by dceara — APPROVED** [review]
Looks good to me, thanks!

**Review by jcaamano — APPROVED** [review]

**Comment by tssurya** [comment]
@kyrtapz can you please open a backport to release-1.0 as well?

**Comment by kyrtapz** [comment]
> @kyrtapz can you please open a backport to release-1.0 as well?

https://github.com/ovn-org/ovn-kubernetes/pull/4742

**Pull request merged**